runtime/vm/vmprops.c

```diff
@@ -152,6 +152,11 @@ getOptionArg(J9JavaVM *vm, IDATA argIndex, UDATA optionNameLen)
 	}
 
 _end:
+	if (NULL != optionArg) {
+		/* Convert the value string from the platform encoding to the modified UTF8 */
+		optionArg = (char *)getMUtf8String(vm, optionArg, strlen(optionArg));
```

Comment:
The places that use this need to pass the J9SYSPROP_FLAG_VALUE_ALLOCATED flag so the memory is freed.

Reply:
and addPropertyForOptionWithModuleListArg() should free the memory.

Reply:
Also, pls add a function comment for this function which mentions the return value is allocated and needs to be freed.

Reply:
Updated the code against the suggestions above.

Reply:
addPropertyForOptionWithModuleListArg() needs to free the memory, and addPropertiesForOptionWithAssignArg needs to pass J9SYSPROP_FLAG_VALUE_ALLOCATED as well.

Reply:
For `addPropertiesForOptionWithAssignArg`, the flag is already in there (not in our fix)
```
addPropertiesForOptionWithAssignArg(J9JavaVM *vm, const char *optionName, UDATA optionNameLen, const char *basePropName, UDATA basePropNameLen, UDATA *propertyCount)
{
				rc = addSystemProperty(vm, propName, optionArg, J9SYSPROP_FLAG_NAME_ALLOCATED); <----
				if (J9SYSPROP_ERROR_NONE != rc) {
					goto _end;
				}

```

Reply:
Added code `j9mem_free_memory(optionArg)` in the loop in `addPropertyForOptionWithModuleListArg()`.

Reply:
add J9SYSPROP_FLAG_VALUE_ALLOCATED to `addPropertiesForOptionWithAssignArg` given only `J9SYSPROP_FLAG_NAME_ALLOCATED` is passed to `addSystemProperty` in there.

Reply:
Ok, and pls add a function comment for this function which mentions the return value is allocated and needs to be freed.

Reply:
Added comment for `optionArg` in `addPropertiesForOptionWithAssignArg`.

Reply:
I'm asking for a function comment to be added to getOptionArg().

Reply:
Added the function comment to `getOptionArg()` as suggested above.